### ISSUE TITLE
CoverInitialisedName parser issues

### ISSUE THREAD

**Issue by anba** [google/traceur-compiler#478]
There are multiple issues when processing the CoverInitialisedName production rule. Test cases:

```
// causes "unreachable" internal error
function f() {
  (1 ? ({a=0}) => {} : 1);
}

// causes "unreachable" internal error
function f() {
  ({a = (0, {a = 0})} = {})
}

// does not report SyntaxError, generates invalid code
function f() {
  (x => {});
  ({a = 0});
}

// generates invalid code
function f() {
  ({a = ({a = 0} = {})} = {});
}
```
